src/internal.coffee

```diff
@@ -14,6 +20,85 @@ module.exports =
       throw new Error "Neither Blob nor Buffer are accessible in this environment. " +
         "Consider adding Blob.js shim"
 
+  generateMHTDocument: (htmlSource) ->
+    # take care of images
+    @prepareImageParts(htmlSource).then (result) ->
+      if result.htmlSource
+        {htmlSource, imageContentParts} = result
+      else
+        htmlSource = htmlSource.outerHTML
+        imageContentParts = result.imageContentParts
+      # for proper MHT parsing all '=' signs in html tags need to be replaced with '=3D'
+      htmlSource = htmlSource.replace /\=/g, '=3D'
+      W.resolve mhtDocumentTemplate {htmlSource, contentParts: imageContentParts.join '\n'}
+    , (err) -> throw err
+
+  prepareImageParts: (htmlSource) ->
+    imageContentParts = []
+    # regular expressions to work with DATA URI sourced (inlined) and regular images
+    inlinedSrcPattern = /"data:(\w+\/\w+);(\w+),(\S+)"/g
+    regularSrcPattern = /(<img[^<>]+src=")(?!file)(?!data\:\S+)(\S+)(")/g
+    # replacer function for images sources via DATA URI
+    inlinedReplacer = (match, contentType, contentEncoding, encodedContent) ->
+      index = imageContentParts.length
+      extension = contentType.split('/')[1]
+      contentLocation = "file://fake/image#{index}.#{extension}"
+      imageContentParts.push mhtPartTemplate {contentType, contentEncoding, contentLocation, encodedContent}
+      "\"#{contentLocation}\""
+
+    if typeof htmlSource is 'string'
+      # don't have to do anything if there are no images in provided htmlSource
+      return W.resolve({htmlSource, imageContentParts}) unless /<img/g.test htmlSource
+      # take care of 'regular' images first
+      matches = htmlSource.match regularSrcPattern
+      if matches
+        deferred = W.defer()
+        # temporary canvas and image objects needed to convert image into base64 string
+        canvas = document.createElement 'canvas'
```

Comment:
This won't work in node.js for obvious reasons. We say that the library is also working server-side, we need to think whether to drop node.js support or work it around.